### PR TITLE
Nicer default meeting theme

### PR DESCRIPTION
**Pull request by pferreir**

### CODE REVIEW

File: indico/htdocs/js/indico/jquery/declarative.js [hunk 1]
```diff
@@ -201,4 +202,15 @@
             $elems.mathJax();
         }
     }
+
+    function setupAnchorLinks() {
+        $('[data-anchor]').each(function(_, elem) {
+            var $elem = $(elem),
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/jquery/declarative.js]: `$(this)`; no need to use any function arguments in that case

File: indico/htdocs/js/indico/jquery/declarative.js [hunk 2]
```diff
@@ -201,4 +202,15 @@
...
+                fragment = $elem.data('anchor');
+            $elem.append($('<a class="anchor-link">&para;</a>')
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/jquery/declarative.js]: `$('<a>', {class: ..., href: ..., title: ...})`

File: indico/htdocs/sass/themes/indico.scss [hunk 3]
```diff
@@ -403,3 +475,43 @@ a.lecture-series-link {
...
+    .timetable-time.nested {
+        border-color: $selected-color;
+
+        .start-time {
+            background-color: $selected-color;
+        }
+    }
+
+    &.timetable-block > .timetable-time .start-time {
+            @include ribbon($selected-color);
```
Comment (ThiefMaster) [inline indico/htdocs/sass/themes/indico.scss]: wrong indentation

File: indico/modules/events/timetable/templates/display/indico/_contribution.html [hunk 4]
```diff
@@ -5,15 +5,19 @@
                                                               render_notes, render_description, render_time %}
 
 {% macro render_contribution(contrib, event, parent=none, nested=false, hide_end_time=false, timezone=none,
-                             show_notes=false) -%}
-    <li class="timetable-item timetable-contrib">
+                             show_notes=false, show_location=false) -%}
+     {% set anchor -%}
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/indico/_contribution.html]: I'd probably go for this:

    {% set anchor = 'c-%d-%s' | format(contrib.friendly_id, contrib.title|slugifx(maxlen=30)) %}

File: indico/util/string.py [hunk 5]
```diff
@@ -129,20 +129,28 @@ def strict_unicode(value):
     return unicode(value)
 
 
-def slugify(value, lower=True):
+def slugify(value, lower=True, maxlen=0):
```
Comment (ThiefMaster) [inline indico/util/string.py]: default to None instead?

File: indico/modules/events/timetable/templates/display/indico/_subcontribution.html [hunk 6]
```diff
@@ -4,10 +4,16 @@
 
 
 {% macro render_subcontribution(subcontrib, event, show_notes=true) %}
-    <li class="timetable-subcontrib timetable-item">
+    {% set contrib = subcontrib.contribution %}
+    {% set anchor -%}
+        sc-{{ contrib.friendly_id }}-
+        {{- subcontrib.friendly_id }}-{{ subcontrib.title | slugify(maxlen=30) }}
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/indico/_subcontribution.html]: I wonder if it'd make sense to accept `*values` in slugify so you could do this:

    {% set anchor = slugify('sc', contrib.friendly_id, 'subcontrib.friendly_id', subcontrib.title, maxlen=30) %}

File: indico/util/string.py [hunk 7]
```diff
@@ -129,20 +129,28 @@ def strict_unicode(value):
...
     :param lower: Whether the slug should be all-lowercase
+    :param maxlen: Maximum slug length
     """
     value = to_unicode(value).encode('translit/long')
     value = unicode(re.sub('[^\w\s-]', '', value).strip())
+
     if lower:
         value = value.lower()
-    return re.sub('[-\s]+', '-', value)
+
```
Comment (ThiefMaster) [inline indico/util/string.py]: wow, much whitespace, such blanks

File: indico/util/string.py [hunk 8]
```diff
@@ -129,20 +129,28 @@ def strict_unicode(value):
...
+    value = re.sub('[-\s]+', '-', value)
```
Comment (ThiefMaster) [inline indico/util/string.py]: `r'[-\s]+'`

File: indico/util/string.py (hunk 5 above)
Comment (pferreir) [inline indico/util/string.py]: Isn't `0` more indicative of the type of that var?

Comment (ThiefMaster) [inline indico/util/string.py]: True, even though a "length" is obviously a number. My thought was that a length of 0 and no length (None) are different things, even though it's common sense that you never want a zero-length slug.

**Pull request merged**